dask_jobqueue/core.py

```diff
@@ -96,7 +173,10 @@ def __init__(self,
             raise NotImplementedError('JobQueueCluster is an abstract class '
                                       'that should not be instanciated.')
 
-        #This attribute should be overriden
+        if '-' in name:
+            raise ValueError('name (%s) can not include the `-` character')
```

Comment:
Is this because of the naming convention we're using to grab JOB-IDs?

Reply:
yes, the tricky part is that sometimes the worker names also have an additional `-integer` when we have grouped workers. For now, I just made the straightforward decision to keep it simple. If there are strong opposing thoughts, I'm happy to change this.

Reply:
I wonder if we could further isolate the name by using something like a double-hyphen

    name='dask-worker--{SGE_JOB_ID}`

That might be distinct enough so that we can isolate the job_id more reliably.